### PR TITLE
Added PR Template and update docs

### PR DESCRIPTION
**Pull request by ZeroX-DG**
## Description

This PR added a pull request template and update docs to keep it up to date

## Issue fixed

https://github.com/BoostIO/Boostnote/issues/2528

## Type of changes

- :white_circle: Bug fix (Change that fixed an issue)
- :white_circle: Breaking change (Change that can cause existing functionality to change)
- :radio_button: Improvement (Change that improves the code. Maybe performance or development improvement)
- :white_circle: Feature (Change that adds new functionality)
- :radio_button: Documentation change (Change that modifies documentation. Maybe typo fixes)

## Checklist:

- :white_circle: My code follows the project code style
- :white_circle: I have written test for my code and it has been tested
- :radio_button: All existing tests have been passed
- :white_circle: I have attached a screenshot/video to visualize my change if possible

### CODE REVIEW

**Comment by ZeroX-DG** [comment]
Yeah, I also feel a bit annoying :angry: What should I do? Do you have any idea @daiyam

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+## Issue fixed
+<!--
+Please list out all issue fixed with this PR here.
+-->
+
+<!--
+Please make sure you fill in these checkboxes,
+your PR will be reviewed faster if we know exactly what it does.
+-->
```
Comment (daiyam) [inline PULL_REQUEST_TEMPLATE.md]: @ZeroX-DG can you add the following text?
```
Put an `x` in all the boxes that apply.
```

Comment (ZeroX-DG) [reply PULL_REQUEST_TEMPLATE.md]: @daiyam Thanks for the feedback, I've updated the message. Do you have any other idea?

**Comment by daiyam** [comment]
@ZeroX-DG I have removed the `- ` before the checkboxes in the PR #2506. It doesn't show up as a task list but it's still readable.

**Comment by ZeroX-DG** [comment]
@daiyam It looks ugly :cry: Is there any other way?

**Comment by daiyam** [comment]
Maybe with emoticon:

- :radio_button: Bug fix (Change that fixed an issue)
- :white_circle: Breaking change (Change that can cause existing functionality to change)
- :white_circle: Improvement (Change that improves the code. Maybe performance or development improvement)
- :white_circle: Feature (Change that adds new functionality)
- :white_circle: Documentation change (Change that modifies documentation. Maybe typo fixes)

```
- :radio_button: Bug fix (Change that fixed an issue)
- :white_circle: Breaking change (Change that can cause existing functionality to change)
- :white_circle: Improvement (Change that improves the code. Maybe performance or development improvement)
- :white_circle: Feature (Change that adds new functionality)
- :white_circle: Documentation change (Change that modifies documentation. Maybe typo fixes)
```

**Comment by ZeroX-DG** [comment]
Oh, that's a brilliant idea! I'll update it soon

**Comment by jangtsekiang** [comment]
@ZeroX-DG , 
Can you create a link for "My code follows the project code style"?

**Comment by ZeroX-DG** [comment]
@jangtsekiang I have added a link and created a doc for code style in this PR.

**Comment by daiyam** [comment]
When will it be merged?

**Comment by ZeroX-DG** [comment]
I'm not sure, @Rokt33r is pretty busy.

**Pull request merged**